nlp/gender_debiasing_cda/gender_debiasing_cda.ipynb

```diff
@@ -0,0 +1,1032 @@
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "nkoNfTSd21g6"
+      },
+      "source": [
+        "## 🛠️ Getting started\n",
+        "\n",
+        "The cells below will configure everything that is required to get started with data loading using HuggingFace and training word2vec models using Gensim."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "LQXfI15jQb1C"
+      },
+      "source": [
+        "### Setup"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "J8uFMXmmHzUY"
+      },
+      "outputs": [],
+      "source": [
+        "!apt-get install python3-magic\n",
+        "!pip install -q gensim==4.1.2 datasets augly gender-bender"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "BTGe29mcQe-7"
+      },
+      "source": [
+        "### Imports"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 1,
+      "metadata": {
+        "id": "iLiKoNuq1MLn"
+      },
+      "outputs": [],
+      "source": [
+        "import numpy as np\n",
+        "from scipy import spatial\n",
+        "import multiprocessing\n",
+        "from datetime import datetime\n",
+        "import pandas as pd\n",
+        "\n",
+        "import plotly.graph_objects as go\n",
+        "\n",
+        "import augly.text as textaugs\n",
+        "from gender_bender import gender_bend\n",
+        "from sklearn.decomposition import PCA\n",
+        "\n",
+        "from datasets import concatenate_datasets, Dataset, load_from_disk"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ipA94F-h01Px"
+      },
+      "source": [
+        "### Download dataset\n",
+        "\n",
+        "We'll train our word vectors on the [Wiki-Bio](https://rlebret.github.io/wikipedia-biography-dataset/) dataset, which is a collection various biography pages from Wikipedia. This seems like an ideal candidate for our gender-bias experiment.\n",
+        "\n",
+        "We pre-downloaded the dataset from the Github page. If you are unable to do so, feel free to just load in our preprocessed dataset in the section **Load in the data**."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "CdIlM9fPT4SS"
+      },
+      "outputs": [],
+      "source": [
+        "wiki_bio_sent = []\n",
+        "wiki_bio_nb = []\n",
+        "wiki_bio_title = []\n",
+        "wiki_bio_id = []\n",
+        "\n",
+        "with open('wikipedia-biography-dataset/test/test.sent', 'r') as f:\n",
+        "    wiki_bio_sent = f.read().splitlines() \n",
+        "\n",
+        "with open('wikipedia-biography-dataset/test/test.nb', 'r') as f:\n",
+        "    wiki_bio_nb = f.read().splitlines()\n",
```

Comment:
If I understand this correctly this only contains ints, instead of converting it below you could to it here already:

`wiki_bio_nr = [int(i) for i in f.readlines()]`

Reply:
True, but now it's more uniform over all file reads, so gonna keep it as is